Source/FieldSolver/SpectralSolver/SpectralSolver.cpp

```diff
@@ -41,32 +41,35 @@ SpectralSolver::SpectralSolver(
     // as well as the value of the corresponding k coordinates)
     const SpectralKSpace k_space= SpectralKSpace(realspace_ba, dm, dx);
 
+    m_spectral_index = SpectralFieldIndex(update_with_rho, fft_do_time_averaging,
+                                          J_linear_in_time, dive_cleaning, divb_cleaning, pml);
+
     // - Select the algorithm depending on the input parameters
     //   Initialize the corresponding coefficients over k space
 
     if (pml) {
         algorithm = std::make_unique<PMLPsatdAlgorithm>(
-            k_space, dm, norder_x, norder_y, norder_z, nodal,
+            k_space, dm, m_spectral_index, norder_x, norder_y, norder_z, nodal,
             fill_guards, dt, dive_cleaning, divb_cleaning);
     }
     else {
         // Comoving PSATD algorithm
         if (v_comoving[0] != 0. || v_comoving[1] != 0. || v_comoving[2] != 0.) {
             algorithm = std::make_unique<ComovingPsatdAlgorithm>(
-                k_space, dm, norder_x, norder_y, norder_z, nodal,
+                k_space, dm, m_spectral_index, norder_x, norder_y, norder_z, nodal,
                 fill_guards, v_comoving, dt, update_with_rho);
         }
         // PSATD algorithms: standard, Galilean, or averaged Galilean
         else {
             algorithm = std::make_unique<PsatdAlgorithm>(
-                k_space, dm, norder_x, norder_y, norder_z, nodal, fill_guards,
+                k_space, dm, m_spectral_index, norder_x, norder_y, norder_z, nodal, fill_guards,
                 v_galilean, dt, update_with_rho, fft_do_time_averaging, J_linear_in_time);
         }
     }
 
     // - Initialize arrays for fields in spectral space + FFT plans
-    field_data = SpectralFieldData( lev, realspace_ba, k_space, dm,
-                    algorithm->getRequiredNumberOfFields(), periodic_single_box);
```

Comment:
With this change, I don't think that the function `getRequiredNumberOfFields` is used anywhere in the code anymore. Therefore, I would suggest to simply remove it from the different spectral classes.

Reply:
Great point, I did not notice that this is not used anymore. Will remove the functions from the various spectral classes.